### PR TITLE
Refactor Customer & CustomerRepository for Spec Harness Compliance

### LINKED ISSUE #91: Refactor Customer & CustomerRepo for Spec Harness Compliance
We need to run the spec harness for these classes and make necessary updates to ensure compliance.

### PR DESCRIPTION
**Pull request by mkrumholz**
## Changes

- Really small changes, just updating formatting for times (allowing for Time or String inputs) and IDs to integers.

## Checklist 
-[x] Tests pass
-[x] Ready to merge

closes #91

### CODE REVIEW

File: lib/customer.rb [hunk 1]
```diff
@@ -34,4 +32,16 @@ def update_last_name(name)
   def update_time
     @updated_at = Time.now
   end
+
+  def created_at
+    return @created_at if @created_at.instance_of?(Time)
+
```
Comment (sami-p) [inline lib/customer.rb:38]: Looks great!! 😊  (Also reminder for later: possibly will remove extra line here)

File: lib/customer.rb [hunk 2]
```diff
@@ -34,4 +32,16 @@ def update_last_name(name)
...
+    Time.parse(@created_at)
+  end
+
+  def updated_at
+    return @updated_at if @updated_at.instance_of?(Time)
+
```
Comment (sami-p) [inline lib/customer.rb:44]: (same reminder)

**Review by sami-p — APPROVED** [review]
¡Que bien! Buen trabajo. Aprobado ✅

**Review by NoahZinter — APPROVED** [review]

File: lib/customer.rb (hunk 1 above)
Comment (mkrumholz) [reply lib/customer.rb:38]: Good reminder! Making that update

**Pull request merged**